django/db/models/sql/query.py

```diff
@@ -800,8 +800,8 @@ def change_aliases(self, change_map):
         # "group by" and "where".
         self.where.relabel_aliases(change_map)
         if isinstance(self.group_by, tuple):
-            self.group_by = tuple([col.relabeled_clone(change_map) for col in self.group_by])
-        self.select = tuple([col.relabeled_clone(change_map) for col in self.select])
+            self.group_by = tuple(col.relabeled_clone(change_map) for col in self.group_by)
```

Comment:
Oh wow. That is surprising and counter to my intuition. Thanks for pointing that out. Judging by the `timeit` output, I suppose the guideline ought to be that `tuple(<list comprehension>)` is normally preferred over `tuple(<generator expression>)`. I find them both very readable, but one is ever slightly faster. What do you think?